library/core/benches/str/corpora.rs

```diff
@@ -0,0 +1,83 @@
+//! Exposes a number of modules with different kinds of strings.
+//!
+//! Each module contains `&str` constants named `SMALL`, `MEDIUM`, `LARGE`, and
+//! `HUGE`.
+//!
+//! - The `SMALL` string is generally around 30-40 bytes.
+//! - The `MEDIUM` string is generally around 600-700 bytes.
+//! - The `LARGE` string is the `MEDIUM` string repeated 8x, and is around 5kb.
+//! - The `HUGE` string is the `LARGE` string repeated 8x (or the `MEDIUM`
+//!   string repeated 64x), and is around 40kb.
+//!
+//! Except for `mod emoji` (which is just a bunch of emoji), the strings were
+//! pulled from (localizations of) rust-lang.org.
+
+macro_rules! repeat8 {
+    ($s:expr) => {
+        concat!($s, $s, $s, $s, $s, $s, $s, $s)
+    };
+}
+
+macro_rules! define_consts {
+    ($s:literal) => {
+        pub const MEDIUM: &str = $s;
+        pub const LARGE: &str = repeat8!($s);
+        pub const HUGE: &str = repeat8!(repeat8!(repeat8!($s)));
+    };
+}
+
+pub mod en {
+    pub const SMALL: &str = "Mary had a little lamb, Little lamb";
```

Comment:
Small might still be too big. E.g. if you're calculating word lengths on some dictionary after splitting the dictionary file by newlines then many words will end up being shorter than a usize or when they're misaligned you'll only get a head and tail and no body.

Reply:
Hm, sure. I'll add a TINY.

Reply:
I've added a tiny. It was about twice as slow for strings that small, so I've added an explicit path for anything less than `INNER_UNROLL*size_of::<usize>()` in length, where it just uses the old version.